front/src/app/components/common/toggle-button/types.ts

```diff
@@ -0,0 +1,8 @@
+export interface IToggle<TValue> {
+    className?: string;
+    title?: string;
+    groupName?: string;
+    value?: TValue;
```

Comment:
this is redundant parameter. 
html forms logic is not applicable for React.js

which use case are you assume for this parameter? there are only two props important for using IToggle as standalone component: Its name and its "checked" state.
So you can replace two props (`value?: TValue; checked?: boolean`) with one: `value: boolean`. Please refer to comment about universal input component api.